pkg/component/pushed_component.go

```diff
@@ -92,24 +93,16 @@ func (d defaultPushedComponent) GetURLs() ([]url.URL, error) {
 	return d.urls, nil
 }
 
+// GetStorage gets the storage using the storage client of the given pushed component
 func (d defaultPushedComponent) GetStorage() ([]storage.Storage, error) {
 	if d.storage == nil {
-		var storageItems []storage.Storage
-		if _, ok := d.provider.(*s2iComponent); ok {
-			storageList, err := storage.ListMounted(d.client, d.GetName(), d.GetApplication())
-			if err != nil {
-				return nil, err
-			}
-			storageItems = append(storageItems, storageList.Items...)
-		}
 		if _, ok := d.provider.(*devfileComponent); ok {
-			storageList, err := storage.DevfileListMounted(d.client.GetKubeClient(), d.GetName(), d.GetApplication())
+			storageList, err := d.storageClient.ListFromCluster()
 			if err != nil {
 				return nil, err
 			}
-			storageItems = append(storageItems, storageList.Items...)
+			d.storage = append(d.storage, storageList.Items...)
```

Comment:
`d.storage` is now nil at this point

```suggestion
			return storageList.Items
```

Reply:
It gets appended successfully, without a panic, even if the value is nil.